soccer_ipm/soccer_ipm/msgs/ball.py

```diff
@@ -0,0 +1,53 @@
+from ipm_interfaces.msg import Point2DStamped
+from ipm_library.exceptions import NoIntersectionError
+from ipm_library.ipm import IPM
+from rclpy.impl.rcutils_logger import RcutilsLogger
+from soccer_ipm.utils import create_field_plane
+import soccer_vision_2d_msgs.msg as sv2dm
+import soccer_vision_3d_msgs.msg as sv3dm
+
+
+def map_ball_array(
+        msg: sv2dm.BallArray,
+        ipm: IPM,
+        output_frame: str,
+        logger: RcutilsLogger,
+        ball_diameter: float) -> sv3dm.BallArray:
+    """
+    Map a given array of 2D ball detections onto the field plane.
+
+    :param msg: The 2D message that should be mapped
+    :param ipm: An instance of the IPM mapping utility
+    :param output_frame: The tf frame of the field
```

Comment:
``output_frame`` isn't the tf frame of the field. It's "which frame_id we want the output to be in".
This applies to the other files too.

Reply:
The soccer ipm maps vision objects onto the field plane, therefore this should be the frame of the field.

Reply:
I think you're right, my mistake.